### ISSUE TITLE
Find board origin and crop out empty space

### ISSUE THREAD

**Issue by FFY00** [yaqwsx/PcbDraw#41]
Hey, currently pcbdraw only crops out at the vertical bottom limit of the board. Would it be possible to do the same in the other 3 side?

Example:
![luna_rev0](https://user-images.githubusercontent.com/11718923/87204769-8e694700-c2fd-11ea-9520-9aba9aa0e4cb.png)

**Comment by set-soft** [comment]
Nice board!
Did you try with this:
https://github.com/yaqwsx/PcbDraw-Lib/blob/master/scrips/resize_canvas.sh

**Comment by FFY00** [comment]
It is! I did not design it :sweat_smile:.

I haven't tried that, I will have a look tomorrow. It would still be cool to have it automatically done, would simply populate and such.

**Comment by yaqwsx** [comment]
Currently, PcbDraw uses a board bounding box reported by KiCAD. This bounding box includes all items in the drawing (even the ones not used by PcbDraw). Cropping to only visible items makes sense. There are three ways to implement it:
- crawl all board items, filter only the visible ones and compute their bounding box - probably a good compromise between the complexity of implementation and dependencies,
- crawl generated SVG and crop it (probably hardest to implement, most robust),
- use Inkscape to crop it (basically what the script mentioned above does) - simplest to implement, but brings another dependency on Inkscape.

I will think about it and try to implement it. Meanwhile, feel free to use the script above.

**Comment by set-soft** [comment]
> I will think about it and try to implement it. Meanwhile, feel free to use the script above.

Please try to avoid the Inkscape dependency, is 172 MiB, is ok for my desktop, but an overkill for a docker image.
I already avoided it for SVG conversions (is the default for `convert` tool).

**Comment by FFY00** [comment]
I am not really aware of the API pcbnew exposes, but if you can iterate through the items and get the origin and size, it should be fairly straight forward to calculate the min and max x and y.

If this is difficult to do with the available API, you could pull inkscape as an *optional* dependency, disabling the functionality when it is missing.

**Comment by yaqwsx** [comment]
Don't worry, I do not plan to use Inkscape as it is a heavy dependency.

**Comment by yaqwsx** [comment]
Today I accidentally found [svgelements](https://github.com/meerk40t/svgelements). From the brief look, it looks exactly like what I need. It would also allow for dropping the `SvgPathItem` and replacing it with something more robust.

**Comment by joaoantoniocardoso** [comment]
Just to mention: that inkscape script doesn't work here, I had to use:
```
inkscape -D --export-filename=out.svg --verb=FileQuit in.svg
```

And yeah, inkscape is a big dependency...

**Comment by yaqwsx** [comment]
This was addressed in 27dafbc6acb31fb294cf749b3d06ee73f185ea6a
